### ISSUE TITLE
Unable to call lein zprint

### ISSUE THREAD

**Issue by denisidoro** [kkinnear/lein-zprint#4]
```
λ lein zprint 
clojure.lang.Compiler$CompilerException: java.lang.ExceptionInInitializerError, compiling:(zprint/spec.cljc:1:1)
 at clojure.lang.Compiler.load (Compiler.java:7391)
    clojure.lang.RT.loadResourceScript (RT.java:372)
    ....
    clojure.main.main (main.java:37)
Caused by: java.lang.IllegalStateException: Attempting to call unbound fn: #'clojure.core/ident?
 at clojure.lang.Var$Unbound.throwArity (Var.java:43)
```

I'm using https://github.com/RickMoynihan/lein-tools-deps and Clojure 1.9.0

**Comment by kkinnear** [comment]
I cannot reproduce this problem with the information that you have given me above.  If you include your `deps.edn` file and your `project.clj` file, I will have at least a chance of reproducing this and figuring out what is wrong.  It it also important to have the output of `lein version`.

I have certainly seen this error when I have misconfigured things, though just now I can't seem to misconfigure them in the right way to cause this to happen.  Even if I could figure out how I did it, that may have nothing to do with your situation.  So I would really like to get more information to let me reproduce (and then fix) this problem.

One thing I did notice --  when you type `lein zprint`, the `lein-zprint` plugin expects a file specification.  Something like `lein zprint src/stuff/*.clj` if your project is called "stuff".  That said, when I type `lein zprint` without a file specification, I don't get an exception -- nothing happens. So, while you will need to put a file specification after `lein zprint` to get anything useful to happen, I do not believe that is what is causing the exception concerning `ident?`.

**Comment by denisidoro** [comment]
Thanks for the quick response!
Here is a minimal repository: https://github.com/denisidoro/lein-zprint-min-error
Leiningen 2.8.1 on Java 1.8.0_151 OpenJDK 64-Bit Server VM

**Comment by kkinnear** [comment]
It turns out that it is a known problem that leiningen plugins can't use Clojure 1.9 features, as this [issue](https://github.com/technomancy/leiningen/issues/2433) notes.  I don't expect that to change anytime soon, judging from the dates in that issue.  zprint itself requires Clojure spec, but lein-zprint gets around that by building with Clojure 1.8 and using `[clojure-future-spec "1.9.0-alpha17"]`, which works.  Another piece of the puzzle is that clojure-future spec can be used as a dependency with Clojure 1.9, and it sort of does not get in the way, in some way that I haven't yet investigated. 

What I believe is happening is that because lein-tools-deps is built with Clojure 1.9, it triggers the part of clojure-future-spec to sort of "not be there" and so zprint doesn't work without access to spec.  I have verified part of this by cloning and building lein-tools-deps with Clojure 1.8, and everything works fine.  I have also built lein-tools-deps with Clojure 1.9 as `:provided`, and at least lein-zprint works fine.  I don't know enough about lein-tools-deps to know if either of these things have broken it or not -- that is, does it depend on Clojure 1.9 or not?

The two paths forward here are to try to get lein-tools-deps to build using Clojure 1.8, and to investigate clojure-future-spec to see if I can figure out how it interacts with Clojure 1.9, and get it to show up anyway -- if, indeed, that is really what is going wrong.  I will pursue both approaches.  To state what I'm sure is obvious, it will be a while before this gets resolved.

**Comment by kkinnear** [comment]
I have found a way around this problem by standing on my head with leiningen.  The disappointing downside is that this approach takes a serious performance hit.  The server.clj file in your example repo takes 5 seconds to format with the regular lein-zprint, and it takes almost 9 seconds with the hoops I have to jump through to get it to work with lein-tools-deps. That is clearly a lot better than not working, but it is a steep price for everyone to pay so that lein-zprint can work with lein-tools-deps.  

I'm thinking of creating another lein-zprint plugin that does the fancy leiningen stuff and works (more slowly) with lein-tools-deps.  Then I have to maintain two of them, which is a pain, but at least people don't all have wait for the addtional seconds every time they format their sources.  Alternatively, I might find some other way to pull this off.  I'm going to continue to work on this a bit over the next week or so, and then decide where to go from there.

**Comment by NoahTheDuke** [comment]
Looks like support for 1.9 in plugins has hit leiningen 2.9.0-SNAPSHOT!

**Comment by kkinnear** [comment]
@NoahTheDuke, that's great to know, thanks!  I'm not entirely sure I understand the implications.  Do you (or does anyone) know if any of the following are true:

1. If someone uses leiningen 2.9.0-SNAPSHOT, then the current version of lein-zprint and lein-tools-deps will work together correctly?  
2. When leiningen 2.9.0 is released, I can produce a version of lein-zprint that uses Clojure 1.9 (instead of the 1.8 as it currently does), and that will work with leiningen and, probably, lein-tools-deps?

I would guess that #2 is true, but I have no idea about #1.  Thanks!

**Comment by kkinnear** [comment]
For several reasons, I am in the process of moving the base Clojure version of zprint from 1.8 to 1.9, and in order to make lein-zprint work right, I also updated leiningen to 2.9.1. One of the reasons was to hopefully resolve this issue. 

When I went to test all of these changes to see if they actually resolved this issue using lein-zprint-min-error, I first tried to reproduce the problem.  I couldn't reproduce the problem until I downgraded leiningen to 2.8.1.  With leiningen 2.8.1 it fails, with 2.9.1 it works -- with no changes to lein-zprint or lien-tools-deps.  So, it looks like you don't need anything different from lein-zprint, if you just upgrade leiningen to 2.9.1 this should all work now.
